core/src/dbs/capabilities.rs

```diff
@@ -190,6 +191,119 @@ impl std::str::FromStr for NetTarget {
 	}
 }
 
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+pub struct MethodTarget {
+	pub method: Method,
+}
+
+// impl display
+impl fmt::Display for MethodTarget {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		write!(f, "{}", self.method.to_str())
+	}
+}
+
+impl Target for MethodTarget {
+	fn matches(&self, elem: &Self) -> bool {
+		self.method == elem.method
+	}
+}
+
+#[derive(Debug)]
+pub struct ParseMethodTargetError;
+
+impl std::error::Error for ParseMethodTargetError {}
+impl fmt::Display for ParseMethodTargetError {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		write!(f, "The provided method target is not a valid RPC method")
+	}
+}
+
+impl std::str::FromStr for MethodTarget {
+	type Err = ParseMethodTargetError;
+
+	fn from_str(s: &str) -> Result<Self, Self::Err> {
+		match Method::parse(s) {
+			Method::Unknown => Err(ParseMethodTargetError),
+			method => Ok(MethodTarget {
+				method,
+			}),
+		}
+	}
+}
+
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+#[non_exhaustive]
+pub enum RouteTarget {
+	Health,
+	Export,
+	Import,
+	Rpc,
+	Version,
+	Sync,
+	Sql,
+	Signin,
+	Signup,
+	Key,
+	Ml,
```

Comment:
should GraphQL be included here?

Reply:
It should! Nice catch, it was not there when I started this PR 😅

Reply:
Included in https://github.com/surrealdb/surrealdb/pull/4512/commits/21ff5140cf991aaa214793e4ad82e4fb3b71425e.